### PR TITLE
SDD-906 - rename label & tests

### PR DESCRIPTION
**Pull request by MrKevJoy**
- Rename label to Data Upload
- Update tests

### CODE REVIEW

**Review by gunndabad — CHANGES_REQUESTED** [review]
These all need to be guarded with a feature flag check

**Review by soharadesign — APPROVED** [review]

File: src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml [hunk 1]
```diff
@@ -41,29 +41,42 @@
                 Venues
             </a>
         </li>
-        @if (Model.ShowCoursesLinks && Model.ShowApprenticeshipsLinks)
+        @if (FeatureFlagProvider.HaveFeature(FeatureFlags.DataManagement) && ((Model.ShowCoursesLinks && Model.ShowApprenticeshipsLinks) || Model.ShowApprenticeshipsLinks || Model.ShowCoursesLinks))
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml:44]: Should be displayed regardless of provider type
```suggestion
        @if (FeatureFlagProvider.HaveFeature(FeatureFlags.DataManagement))
```

Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml]: I've removed these now.

File: src/Dfc.CourseDirectory.WebV2/SharedViews/Components/AdminProviderContextNav.cshtml [hunk 2]
```diff
@@ -41,29 +41,40 @@
                 Venues
             </a>
         </li>
-        @if (Model.ShowCoursesLinks && Model.ShowApprenticeshipsLinks)
+        @if (FeatureFlagProvider.HaveFeature(FeatureFlags.DataManagement))
         {
-            <li class="pttcd-subnav__item@(currentController == "BulkUpload" || currentController == "BulkUploadApprenticeships" ? " govuk-header__navigation-item--active" : null)">
-                <a asp-action="LandingOptions" asp-controller="BulkUpload" class="pttcd-subnav__item__link" data-testid="adminsubnav-bulkupload">
-                    Bulk upload
+            <li class="pttcd-subnav__item@(currentController == "DataManagement" ? " govuk-header__navigation-item--active" : null)">
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/SharedViews/Components/AdminProviderContextNav.cshtml:46]:
```suggestion
            <li class="pttcd-subnav__item@(currentController.StartsWith("DataManagement") ? " govuk-header__navigation-item--active" : null)">
```

File: src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml [hunk 3]
```diff
@@ -41,29 +41,42 @@
                 Venues
             </a>
         </li>
-        @if (Model.ShowCoursesLinks && Model.ShowApprenticeshipsLinks)
+        @if (FeatureFlagProvider.HaveFeature(FeatureFlags.DataManagement))
         {
-            <li class="govuk-header__navigation-item@(currentController == "BulkUpload" || currentController == "BulkUploadApprenticeships" ? " govuk-header__navigation-item--active" : null)">
-                <a asp-action="LandingOptions" asp-controller="BulkUpload" class="govuk-header__link" data-testid="topnav-bulkupload">
-                    Bulk upload
+            <li class="govuk-header__navigation-item@(currentController == "DataManagement" ? " govuk-header__navigation-item--active" : null)">
```
Comment (gunndabad) [inline src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml:46]:
```suggestion
            <li class="govuk-header__navigation-item@(currentController.StartsWith("DataManagement") ? " govuk-header__navigation-item--active" : null)">
```

File: tests/Dfc.CourseDirectory.Testing/appsettings.Testing.json [hunk 4]
```diff
@@ -2,7 +2,7 @@
   "ConnectionStrings": {
     "DefaultConnection": "Data Source=(local);Initial Catalog=CourseDirectoryTests;Integrated Security=yes"
   },
-  "FeatureFlags": "CoursesDataManagement,VenuesDataManagement",
+  "FeatureFlags": "CoursesDataManagement,VenuesDataManagement, DataManagement",
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.Testing/appsettings.Testing.json:5]: This shouldn't be required.

File: tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs [hunk 5]
```diff
@@ -403,17 +404,17 @@ public async Task ProviderUserForFEAndApprenticeshipsAndTLevels_RendersExpectedN
...
                 topLevelLinks[6].TestId.Should().Be("topnav-signout");
             }
 
             subNavLinks.Count.Should().Be(0);
         }
 
         [Theory]
-        [InlineData(ProviderType.Apprenticeships, "/BulkUploadApprenticeships")]
-        [InlineData(ProviderType.FE | ProviderType.Apprenticeships, "/BulkUpload/LandingOptions")]
+        [InlineData(ProviderType.Apprenticeships, "/data-upload")]
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs:416]: This whole test can go I think

File: tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs [hunk 6]
```diff
@@ -432,14 +433,14 @@ public async Task ProviderUserForFEAndApprenticeshipsAndTLevels_RendersExpectedN
...
         }
 
         [Theory]
-        [InlineData(ProviderType.FE, "/BulkUpload")]
-        [InlineData(ProviderType.Apprenticeships, "/BulkUploadApprenticeships")]
-        [InlineData(ProviderType.FE | ProviderType.Apprenticeships, "/BulkUpload/LandingOptions")]
+        [InlineData(ProviderType.FE, "/data-upload")]
+        [InlineData(ProviderType.Apprenticeships, "/data-upload")]
+        [InlineData(ProviderType.FE | ProviderType.Apprenticeships, "/data-upload")]
         public async Task ProviderTopNavBulkUploadLinksAreCorrect(
```
Comment (gunndabad) [inline tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs:444]: This test can go to

File: tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs (hunk 5 above)
Comment (MrKevJoy) [reply tests/Dfc.CourseDirectory.WebV2.Tests/LayoutTests.cs]: done

File: tests/Dfc.CourseDirectory.Testing/appsettings.Testing.json (hunk 4 above)
Comment (MrKevJoy) [reply tests/Dfc.CourseDirectory.Testing/appsettings.Testing.json]: done

File: src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml (hunk 3 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/SharedViews/Components/ProviderTopNav.cshtml]: done

File: src/Dfc.CourseDirectory.WebV2/SharedViews/Components/AdminProviderContextNav.cshtml (hunk 2 above)
Comment (MrKevJoy) [reply src/Dfc.CourseDirectory.WebV2/SharedViews/Components/AdminProviderContextNav.cshtml]: done

**Review by gunndabad — APPROVED** [review]

**Pull request merged**